### PR TITLE
Feature/custom vehicle data 0173

### LINKED ISSUE #116: [SDL 0173] Read Generic Network Signal Data
Proposal: [Read Generic Network Signal Data](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0173-Read-Generic-Network-Signal-data.md)

> This proposal is to make vehicle data APIs (GetVehicleData/SubscribeVehicleData/UnsubscribeVehicleData/OnVehicleData) more flexible to allow additional new vehicle data items, while maintaining the security access via policies.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/502

Steering Committee Decision:

> The Steering Committee voted to accept this proposal with revisions. The author is to add a statement explaining how old core implementations using this method can be updated to expose new standardized vehicle data items. Additionally, the author is to add the three revisions noted in [this comment.](https://github.com/smartdevicelink/sdl_evolution/issues/502#issuecomment-465326262)

>The proposal .md was updated on March 11, 2019 to include the agreed upon revisions.

### PR DESCRIPTION
**Pull request by nickschwab**
Fixes #116 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Complete entire SDL Server test plan spreadsheet.

### Summary
Support for custom vehicle data creating, reading editing, deleting, and promoting, as well as injection of vehicle data in the Policy Table.

##### Breaking Changes
None

##### Enhancements
* New "Custom Vehicle Data" UI and supporting APIs
* Include `vehicle_data` in Policy Table
* Ability to flag Functional Groups as proprietary
* Ability to manually assign proprietary Functional Groups to application versions
* Addition of `custom_vehicle_data_mapping_url` properties to Module Config

##### Bug Fixes
None

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: app/v1/policy/sql.js [hunk 1]
```diff
@@ -58,6 +58,75 @@ function getAppModules (appId) {
...
+
+    if(onlyVehicleData){
+        query.where({
+            'rst.element_type': 'FUNCTION',
+            'rst.name': 'GetVehicleData',
+            'rst.message_type': 'response'
+        });
+    }
+
+    console.log(query.toString());
```
Comment (nickschwab) [inline app/v1/policy/sql.js]: @nickschwab delete this.

File: src/components/VehicleData.vue [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                            min_size: item.min_size,
+                            max_size: item.max_size,
+                            min_value: item.min_value,
+                            max_value: item.max_value,
+                            array: item.array,
+                            is_deleted: item.is_deleted,
+                            created_ts: item.created_ts,
+                            updated_ts: item.updated_ts,
+                            params: item.params
+                        }"
```
Comment (crokita) [inline src/components/VehicleData.vue]: Try to reduce number of lines using Object.assign

File: app/v1/vehicle-data/helper.js [hunk 3]
```diff
@@ -0,0 +1,659 @@
...
+
+        if (err) {
+            if (err.skipReason) {
+                //warning only, spec already imported etc.
+                app.locals.log.info(err.skipReason);
+            } else {
+                app.locals.log.error(err);
+            }
+        } else {
+
```
Comment (nickschwab) [inline app/v1/vehicle-data/helper.js]:
```suggestion
```

File: app/v1/vehicle-data/helper.js [hunk 4]
```diff
@@ -0,0 +1,659 @@
...
+            app.locals.log.info('Rpc spec updated');
```
Comment (nickschwab) [inline app/v1/vehicle-data/helper.js]: Add cache removal of `vehicleData` upon successful sync of RPC Spec

Comment (nickschwab) [inline app/v1/vehicle-data/helper.js]:
```suggestion
            app.locals.log.info('New RPC Spec saved');
```

File: app/v1/groups/sql.js [hunk 5]
```diff
@@ -88,13 +98,68 @@ function getFuncGroupHmiLevelsId (id) {
...
+            .from('view_custom_vehicle_data vcvd')
+            .where({
+                'vcvd.parent_id': null,
+                'vcvd.is_deleted': false,
+                'vcvd.name': sql('fgp.parameter')
+            })
+            .where(
+                sql.or(
+                    {
+                        'vcvd.status': 'PRODUCTION'
```
Comment (nickschwab) [inline app/v1/groups/sql.js]: Add comment about what this is doing

**Pull request merged**